coding-tracker/coding-tracker/Controller.cs

```diff
@@ -0,0 +1,143 @@
+﻿using System.Configuration;
+using Microsoft.Data.Sqlite;
+using Dapper;
+
+namespace coding_tracker
+{
+    internal class Controller
+    {
+        string connectionString = ConfigurationManager.AppSettings.Get("ConnectionString") + ConfigurationManager.AppSettings.Get("DatabasePath");
+
+        public void InitialiseDatabase()
+        {
+            using (var connection = new SqliteConnection(connectionString))
+            {
+                connection.Open();
+
+                var tableCMD = connection.CreateCommand();
+
+                tableCMD.CommandText = @"CREATE TABLE IF NOT EXISTS coding_time (
+                                           Id INTEGER PRIMARY KEY AUTOINCREMENT,
+                                           Date TEXT,
+                                           StartTime TEXT,
+                                           EndTime TEXT,
+                                           Duration INTEGER)";
+
+                tableCMD.ExecuteNonQuery();
+            }
+        }
+
+        public void Delete()
+        {
+            Console.Clear();
+            ShowAllRecords();
+
+            // take user input (number)
+            var recordId = UserInput.GetNumberInput("Please input the record id number that you would like to delete.");
```

Comment:
🟠 **Separation of Concerns**

💡 You have tightly coupled your Database Controller with your User Input. You could have done you user input first, then called this method with the `recordId`. and then returned a result so that the UI can write the message.

🔧 `public bool Delete(int recordId) { ... return rowsAffected > 0; }`